core/src/main/scala/com/mesosphere/usi/core/FrameEffects.scala

```diff
@@ -0,0 +1,31 @@
+package com.mesosphere.usi.core
+
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+import com.mesosphere.usi.models._
+
+case class FrameEffects(reverseStateEvents: List[USIStateEvent] = Nil, reverseMesosCalls: List[Mesos.Call] = Nil) {
+  lazy val stateEvents = reverseStateEvents.reverse
+  lazy val mesosCalls = reverseMesosCalls.reverse
+
+  def ++(other: FrameEffects): FrameEffects = {
+    if (other == FrameEffects.empty)
+      this
+    else if (this == FrameEffects.empty)
+      other
+    else
+      FrameEffects(other.reverseStateEvents ++ reverseStateEvents, other.reverseMesosCalls ++ reverseMesosCalls)
+  }
+
+  def withPodStatus(id: PodId, newStatus: Option[PodStatus]): FrameEffects =
+    withStateUpdated(PodStatusUpdated(id, newStatus))
+  def withPodRecord(id: PodId, newRecord: Option[PodRecord]): FrameEffects =
```

Comment:
This needs some docs. Eg when `newRecord` is empty it means a deletion, right? Should we have a `withoutPodRecord(id)` as well?

Reply:
`withPodRecord` None handles it for now; I'll add docs